torch/optim/lr_scheduler.py

```diff
@@ -359,6 +380,10 @@ def _get_closed_form_lr(self):
         return [base_lr * self.gamma ** (self.last_epoch // self.step_size)
                 for base_lr in self.base_lrs]
 
+    @property
+    def steps_on(self):
+        return StepsOnFlag.EPOCH
```

Comment:
Thinking out loud: Looking at `StepLR`, I could imagine that users may want to step on after a fix number of iterations, which may or may not be epochs. The property is nice in the sense that it doesn't force people to use it, yet should cover most use case.

Reply:
As discussed in the issue, a few of the schedulers could be leveraged by users in less traditional manners (e.g., every B batches, every E epochs, or based on some conditional logic). As is, I've currently set them to signal the stepping behavior as described in the documentation (at least as of PyTorch 1.4).

For the default schedulers, there's not much we can do other than let the user maintain the non-traditional stepping logic in their codebase and let them ignore the flag. However, if a user is to create their own scheduler, either by inheriting from an existing one or creating their own, they can dynamically set the `steps_on` property using arbitrary internal logic to ensure that code which follows the generic pattern listed below would still work. As an aside, if we wish to enable users to more easily support this use case, we may wish to include a `StepsOnFlag.NONE` to indicate that it should neither step on a batch nor epoch for the time being.

```python
for _ in range(num_epochs):
    for batch in dl:
        train_batch(batch)
        if scheduler.steps_on & StepsOnFlag.BATCH:
            scheduler.step()

    validate()
    if scheduler.steps_on & StepsOnFlag.EPOCH:
        scheduler.step()
```